Source/WebCore/dom/Document.cpp

```diff
@@ -1507,6 +1508,23 @@ void Document::setContentLanguage(const AtomString& language)
     m_styleScope->didChangeStyleSheetEnvironment();
 }
 
+const AtomString& Document::effectiveDocumentElementLanguage() const
+{
+    if (!m_effectiveDocumentElementLanguage.isNull())
+        return m_effectiveDocumentElementLanguage;
+    return m_contentLanguage;
+}
+
+void Document::setEffectiveDocumentElementLanguage(const AtomString& language)
+{
+    if (m_effectiveDocumentElementLanguage == language)
+        return;
+    m_effectiveDocumentElementLanguage = language;
+
+    // Recalculate style so language is used when selecting the initial font.
+    m_styleScope->didChangeStyleSheetEnvironment();
+}
```

Comment:
We can also skip style update if m_effectiveDocumentElementLanguage.isNull() &&  m_contentLanguage ==  language (that is if effectiveDocumentElementLanguage() doesn't change)